sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,24 @@ def reduce_inequalities(inequalities, symbols=[]):
 
     # restore original symbols and return
     return rv.xreplace({v: k for k, v in recast.items()})
+
+
+def reduce_boolean_expr(boolean_expr,symbols=[]):
+    boolean_expr=to_cnf(boolean_expr)
+    if isinstance(boolean_expr, And) and not(boolean_expr.has(Or)):
+          return(reduce_inequalities(boolean_expr.args,symbols))
+    elif boolean_expr.has(Or):
+        common=None
+        for i in boolean_expr.args:
```

Comment:
`boolean_expr` might be an And with one or more args an Or or an Or at this point.